plugins/modules/dcnm_inventory.py

```diff
@@ -74,20 +78,85 @@
         description:
         - Maximum Hops to reach the switch
         type: str
-        required: true
+        required: false
+        default: 0
       role:
         description:
         - Role which needs to be assigned to the switch
         choices: ['leaf', 'spine', 'border', 'border_spine', 'border_gateway', 'border_gateway_spine',
                  'super_spine', 'border_super_spine', 'border_gateway_super_spine']
         type: str
-        required: true
+        required: false
         default: leaf
-      preserve_configs:
+      preserve_config:
         description:
         - Set this to false for greenfield deployment and true for brownfield deployment
-        type: str
-        required: true
+        type: bool
+        required: false
+        default: false
+      poap:
+        description:
+        - Configurations of switch to Bootstrap/Pre-provision.
+          Please note that POAP and DHCP configurations needs to enabled in fabric configuration
```

Comment:
> Also it would be good to provide a task using `dcnm_rest` module before the task that calls `dcnm_inventory` to set the required fabric configs

I had added this task in the example, but it takes more than 300 lines just for that task, so decided against adding it. If it's ok to have a example task with 300 lines, I will add it back

Reply:
Is it because you are including all of the fabric properties in the `dcnm_rest` call?  You could be able to just specify the properties required for POAP and let all of the other properties use the default fabric values.

Reply:
Added an example task in the inventory module as suggested. Added a sample task in IT cases too in the poap test yaml file